build_daemon/lib/data/server_log.dart

```diff
@@ -4,14 +4,47 @@
 
 import 'package:built_value/built_value.dart';
 import 'package:built_value/serializer.dart';
+import 'package:logging/logging.dart';
 
 part 'server_log.g.dart';
 
+/// Roughly matches the `LogRecord` class from `package:logging`.
 abstract class ServerLog implements Built<ServerLog, ServerLogBuilder> {
   static Serializer<ServerLog> get serializer => _$serverLogSerializer;
 
   factory ServerLog([updates(ServerLogBuilder b)]) = _$ServerLog;
 
+  factory ServerLog.fromLogRecord(LogRecord record) => ServerLog((b) => b
+    ..message = record.message
+    ..level = record.level.value
+    ..loggerName = record.loggerName
+    ..error = record?.error?.toString()
+    ..stackTrace =
+        record.stackTrace == null ? null : record.stackTrace.toString());
+
+  LogRecord toLogRecord() {
+    var logLevel = Level.LEVELS.firstWhere((l) => l.value == level,
+        orElse: () => throw ArgumentError.value(
+            level,
+            'level',
+            'Only pre-defined log levels are supported, see [Level.LEVELS] '
+                'from package:logging.'));
+    return LogRecord(logLevel, message, loggerName ?? '', error,
+        stackTrace == null ? null : StackTrace.fromString(stackTrace));
+  }
+
   ServerLog._();
-  String get log;
+
+  int get level;
```

Comment:
Should we make this an enum? This would likely prevent the potential exception you see above.

Reply:
Hmm, that actually might be nice. Then we can add a simple helper here for converting between levels also?

We would still have to throw the exception when converting from a `LogRecord` though, but that is less common.

Reply:
Done